kernel/sql-federation/executor/src/main/java/org/apache/shardingsphere/sqlfederation/executor/enumerator/JDBCRowEnumerator.java

```diff
@@ -64,11 +67,16 @@ private boolean moveNext0() throws SQLException {
     private void setCurrentRow() throws SQLException {
         Object[] rowValues = new Object[metaData.getColumnCount()];
         for (int i = 0; i < metaData.getColumnCount(); i++) {
-            rowValues[i] = queryResult.getValue(i + 1, Object.class);
+            rowValues[i] = getValue(i);
         }
         this.currentRow = 1 == metaData.getColumnCount() ? rowValues[0] : rowValues;
     }
     
+    private Object getValue(final int index) throws SQLException {
```

Comment:
Can you add e2e test for this convert logic?

Reply:
Hi mentor, there is no field related to `datetime` in the e2e standard table.

Reply:
Thank you for your feedback, we could consider add a e2e test table which contains all data type.